紀錄/23 20210323.md

```diff
@@ -0,0 +1,52 @@
+# LeetCode 讀書會第 23 次聚會 2021/03/23
+
+  leetcode 讀書會通知
+ 1. 項目: 第 23  次聚會
+ 2. 目的: 線上一起寫題目, 由有想法的人帶領, 先解題, 再看該題有趣的解法
+ 3. 時間: 3/23 (二) 20:00 ~ 21:00
+ 4. 地點: google meet 線上 (前 10 分鐘預備鏈接)
+ 5. 解題項目:  [Binary search](https://leetcode.com/explore/learn/card/linked-list)
+ 6. 共筆: GitHub https://github.com/programmingbookclub/Leetcode-club
+
+* Background
+*   Article  How does it work?
+*   EASY	 704	 Binary Search	 https://leetcode.com/problems/binary-search
+*   Article  Identification and Template Introduction
+* 
+* Template I
+*   Article Binary Search Template I
+*     EASY	 69	 Sqrt(x)	 https://leetcode.com/problems/sqrtx
+*     EASY	 374	 Guess Number Higher or Lower	 https://leetcode.com/problems/guess-number-higher-or-lower
```

Comment:
```suggestion
*     EASY	 374	 Guess Number Higher or Lower	 https://leetcode.com/problems/guess-number-higher-or-lower

```Swift
func guessNumber(_ n: Int) -> Int {
    var low = 0
    var high = n
    while low <= high {
        let mid = low + (high - low)/2
        let result = guess(mid)
        switch result {
            // picked number is lower
            case -1: high = mid - 1
            // picked number is higher
            case 1: low = mid + 1
            default: return mid
        }
    }
    return 0
}
```

```